db/rdbms/src/main/java/io/camunda/db/rdbms/read/service/VariableReader.java

```diff
@@ -8,17 +8,41 @@
 package io.camunda.db.rdbms.read.service;
 
 import io.camunda.db.rdbms.sql.VariableMapper;
-import io.camunda.db.rdbms.write.domain.VariableDbModel;
+import io.camunda.search.entities.VariableEntity;
+import io.camunda.search.filter.VariableFilter.Builder;
+import io.camunda.search.page.SearchQueryPage;
+import io.camunda.search.query.VariableQuery;
+import io.camunda.search.sort.VariableSort;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class VariableReader {
 
+  private static final Logger LOG = LoggerFactory.getLogger(VariableReader.class);
+
   private final VariableMapper variableMapper;
 
   public VariableReader(final VariableMapper variableMapper) {
     this.variableMapper = variableMapper;
   }
 
-  public VariableDbModel findOne(final Long key) {
-    return variableMapper.findOne(key);
+  public VariableEntity findOne(final Long key) {
+    return search(
+            new VariableQuery(
+                new Builder().variableKeys(key).build(),
+                VariableSort.of(b -> b),
+                SearchQueryPage.of(b -> b.from(0).size(10))))
```

Comment:
❓ Why does `findOne` need to search for a page of 10 records? Can it be just 0,1?

Reply:
You are right!